scapy/fields.py

```diff
@@ -173,7 +176,7 @@ def getfield(self, pkt, s):
 
     def addfield(self, pkt, s, val):
         sval = self._fld.addfield(pkt, "", val)
-        return s+sval+struct.pack("%is" % (self.padlen(len(sval))), self._padwith)
+        return raw(s)+sval+struct.pack("%is" % (self.padlen(len(sval))), raw(self._padwith))
```

Comment:
Are the `raw()` calls needed here? Be careful to call these functions only when necessary, the impact on performances could be important.